### DISCUSSION TITLE
Logic Pro X + Mic -> Skype

### DISCUSSION THREAD

**Discussion by heitorsilva** [ExistentialAudio/BlackHole#107] [General]
Hi @devinroth! Thanks for your work on Blackhole...

I was reading the issues here on Github to see if there was anyone trying to do the same as me, and I think the only one I found was #98, but I'm not sure, so I decided to write my own.

I want to have a Skype call with a person, and be able to route my Logic Pro X audio to the call, while I'm also listening (monitoring) Logic.

I tried the aggregated device, I tried the multi-output, and nothing worked. The only way I managed to route Logic audio to Skype was setting the Output of logic to Blackhole ch16, and then setting the mic of the Skype call to Blackhole ch16 too... but then, I don't listen to Logic and I lose my mic to talk to the person.

This is the latest configuration I tried that didn't work:
<img width="670" alt="Screenshot 2020-04-17 at 03 40 31" src="https://user-images.githubusercontent.com/878699/79522894-3227bb80-805d-11ea-9832-72e92e20e3a5.png">
<img width="668" alt="Screenshot 2020-04-17 at 03 40 48" src="https://user-images.githubusercontent.com/878699/79522912-3bb12380-805d-11ea-8e49-538ade197f33.png">
<img width="759" alt="Screenshot 2020-04-17 at 03 41 19" src="https://user-images.githubusercontent.com/878699/79522952-4ff52080-805d-11ea-9042-bd44fd9bafad.png">
<img width="699" alt="Screenshot 2020-04-17 at 03 42 35" src="https://user-images.githubusercontent.com/878699/79523002-7ca93800-805d-11ea-98c2-c2871fb9e055.png">
<img width="698" alt="Screenshot 2020-04-17 at 03 43 00" src="https://user-images.githubusercontent.com/878699/79523047-964a7f80-805d-11ea-9236-299bc091ecbd.png">

With the configuration above, I'm setting the mic to "Aggregate device" and the person can hear my voice, but doesn't hear Logic, while I'm able to listen to Logic and the person.

If use Loopback, the configuration for what I want is like below:

<img width="698" alt="Screenshot 2020-04-17 at 03 44 42" src="https://user-images.githubusercontent.com/878699/79523149-c85be180-805d-11ea-91bb-7f8e2b919184.png">
<img width="1200" alt="Screenshot 2020-04-17 at 03 45 14" src="https://user-images.githubusercontent.com/878699/79523185-db6eb180-805d-11ea-8ae0-681c765b16e0.png">

With the configuration above, by setting the mic to "Loopback Audio", the person on the other end can hear my mic and Logic, while I'm also able to listen Logic through my iRig output.

How do I replicate that with Blackhole? I'm using Catalina 10.15.4.

**Comment by devinroth** [comment]
You are close. You need to send audio from Logic to channel 1-2 (iRig) to monitor and 3-4 (BlackHole) to Skype. I just setup a send that goes to BlackHole on the tracks I want.

<img width="592" alt="Screen Shot 2020-04-16 at 7 43 11 PM" src="https://user-images.githubusercontent.com/13010961/79526641-97a68880-801a-11ea-944c-7782f41de896.png">

**Comment by devinroth** [comment]
You are close. You need to send audio from Logic to channel 1-2 (iRig) to monitor and 3-4 (BlackHole) to Skype. I just setup a send that goes to BlackHole on the tracks I want.

<img width="592" alt="Screen Shot 2020-04-16 at 7 43 11 PM" src="https://user-images.githubusercontent.com/13010961/79526641-97a68880-801a-11ea-944c-7782f41de896.png">

**Comment by devinroth** [comment]
Alternatively you can make a multi-output and set that as your output on Logic.

**Comment by devinroth** [comment]
Alternatively you can make a multi-output and set that as your output on Logic.

**Comment by heitorsilva** [comment]
@devinroth,

I used BUS 1 to route all my tracks there (to kinda have a master track for the stream), and then I used the send of BUS 1 as you said:

<img width="1440" alt="Screenshot 2020-04-17 at 11 43 28" src="https://user-images.githubusercontent.com/878699/79555949-aa63a080-80a0-11ea-888e-8ae07dd2cb62.png">

<img width="1440" alt="Screenshot 2020-04-17 at 11 44 33" src="https://user-images.githubusercontent.com/878699/79556060-d2530400-80a0-11ea-94fe-88a0084cf2d3.png">

In Skype, I'm using Aggregate Device as the mic... with this configuration, the other side of the conversation still hears my voice but not the music of Logic.

I can hear Logic, talk to the person, hear the person, but the person only hears my voice.
=(

What am I doing wrong? What is missing?

**Comment by heitorsilva** [comment]
@devinroth,

I used BUS 1 to route all my tracks there (to kinda have a master track for the stream), and then I used the send of BUS 1 as you said:

<img width="1440" alt="Screenshot 2020-04-17 at 11 43 28" src="https://user-images.githubusercontent.com/878699/79555949-aa63a080-80a0-11ea-888e-8ae07dd2cb62.png">

<img width="1440" alt="Screenshot 2020-04-17 at 11 44 33" src="https://user-images.githubusercontent.com/878699/79556060-d2530400-80a0-11ea-94fe-88a0084cf2d3.png">

In Skype, I'm using Aggregate Device as the mic... with this configuration, the other side of the conversation still hears my voice but not the music of Logic.

I can hear Logic, talk to the person, hear the person, but the person only hears my voice.
=(

What am I doing wrong? What is missing?

**Comment by devinroth** [comment]
Use only BlackHole as mic into Skype.

**Comment by devinroth** [comment]
Use only BlackHole as mic into Skype.

**Comment by heitorsilva** [comment]
@devinroth If I do that, I can't speak and stream Logic at the same... and that's the whole point of what I want to do... I want to speak and stream Logic to the call.

Thinking logically on the setup, I need to route the Output 3-4 of BH to the Input 4-5 of BH on my setup. Is that possible somehow? Because then, using the Aggregate Device as the mic on Skype will work to talk and hear Logic the same way I do on Loopback.

**Comment by heitorsilva** [comment]
@devinroth If I do that, I can't speak and stream Logic at the same... and that's the whole point of what I want to do... I want to speak and stream Logic to the call.

Thinking logically on the setup, I need to route the Output 3-4 of BH to the Input 4-5 of BH on my setup. Is that possible somehow? Because then, using the Aggregate Device as the mic on Skype will work to talk and hear Logic the same way I do on Loopback.

**Comment by devinroth** [comment]
I think you are misunderstanding how the aggregate works. 

Skype will only take channel 1-2 of whatever the audio input is. That means you need to set it to BlackHole and have your mic and DAW routed to BlackHole 1-2.

If you have the aggregate set in Skype you will only get audio from channels 1/2 of the aggregate. If BlackHole is channels 3-19 you won't hear any of it.

**Comment by devinroth** [comment]
I think you are misunderstanding how the aggregate works. 

Skype will only take channel 1-2 of whatever the audio input is. That means you need to set it to BlackHole and have your mic and DAW routed to BlackHole 1-2.

If you have the aggregate set in Skype you will only get audio from channels 1/2 of the aggregate. If BlackHole is channels 3-19 you won't hear any of it.

**Comment by heitorsilva** [comment]
@devinroth that means I would have to make another software route the input of my mic to the output of BH 1-2, as well as Logic outputting to BH 1-2 at the same time, right? there's no direct configuration on System Preferences or Audio MIDI Setup to route my mic to BH 1-2, is there?

if that is the case, using Loopback is easier (but expensive), because it does what I described above with the virtual device it creates.

**Comment by heitorsilva** [comment]
@devinroth that means I would have to make another software route the input of my mic to the output of BH 1-2, as well as Logic outputting to BH 1-2 at the same time, right? there's no direct configuration on System Preferences or Audio MIDI Setup to route my mic to BH 1-2, is there?

if that is the case, using Loopback is easier (but expensive), because it does what I described above with the virtual device it creates.

**Comment by devinroth** [comment]
I have no issues routing the mic to BH in logic. 

Yes obviously Loopback has more features. BlackHole is only the driver. 

Devin

**Comment by devinroth** [comment]
I have no issues routing the mic to BH in logic. 

Yes obviously Loopback has more features. BlackHole is only the driver. 

Devin

**Comment by heitorsilva** [comment]
All right! Now I get it... thank you very much @devinroth.

Sorry for so many messages. If you want, you can close the issue.

Stay safe...

Cheers

**Comment by heitorsilva** [comment]
All right! Now I get it... thank you very much @devinroth.

Sorry for so many messages. If you want, you can close the issue.

Stay safe...

Cheers

**Comment by devinroth** [comment]
I'll keep it up for a while. Maybe it will help someone else out.

**Comment by devinroth** [comment]
I'll keep it up for a while. Maybe it will help someone else out.

**Comment by lditti** [comment]
Hi @devinroth - thanks for creating this utility,  exactly what I needed. 

I need to send audio coming from Logic and the audio being picked up in the Macbooks internal mic through Skype. I’m using headphones plugged directly into the MacBook to hear logic and the Skype call. 

I tried multi output and the aggregator buy can’t seem to make it work. What would be the simplest way to do this?  I read through the above but was confused. 

Thanks in advance. 
Lino

**Comment by lditti** [comment]
Hi @devinroth - thanks for creating this utility,  exactly what I needed. 

I need to send audio coming from Logic and the audio being picked up in the Macbooks internal mic through Skype. I’m using headphones plugged directly into the MacBook to hear logic and the Skype call. 

I tried multi output and the aggregator buy can’t seem to make it work. What would be the simplest way to do this?  I read through the above but was confused. 

Thanks in advance. 
Lino

**Comment by devinroth** [comment]
I plan on doing a tutorial for this in the nearish future. I’ll post on this thread.

**Comment by devinroth** [comment]
I plan on doing a tutorial for this in the nearish future. I’ll post on this thread.

**Comment by lditti** [comment]
Great - thank you - looking forward to it.

**Comment by lditti** [comment]
Great - thank you - looking forward to it.

**Comment by rmonge** [comment]
I have a partial solution.  I made an aggregate device with BH+ USB interface (for mic input into my daw and midi devices).   I set Aggregate device to both input and output in DAW and Input and Output in Zoom (could be skype).    

Then I set up my daw to capture my mic which was aggregate 17+18 and set the sterio out to be Aggregate 1-2.  This was picked up as the mic input in zoom.   

The trouble I had is that Zoom also used the Aggregate devices as I/O and it writes to same channels that my DAW (logic pro) write out.  

My quick hack was to make my mic the Aggregate 1 channel and then Zoom the aggregate 2 channel (so mono) and it worked.  I could capture the Zoom speaker and send my whole daw output with a mix minus to Zoom w/out the callers output.  It's a bit of a hack since it's mono for each speaker but good enough to get going.     

If there were two BH devices one BH(2) and BH(16) I think this problem would go away since each software wouldn't be sharing the same output devices.   

Thanks for the great tool, can't wait to see more routing features :-)

**Comment by rmonge** [comment]
I have a partial solution.  I made an aggregate device with BH+ USB interface (for mic input into my daw and midi devices).   I set Aggregate device to both input and output in DAW and Input and Output in Zoom (could be skype).    

Then I set up my daw to capture my mic which was aggregate 17+18 and set the sterio out to be Aggregate 1-2.  This was picked up as the mic input in zoom.   

The trouble I had is that Zoom also used the Aggregate devices as I/O and it writes to same channels that my DAW (logic pro) write out.  

My quick hack was to make my mic the Aggregate 1 channel and then Zoom the aggregate 2 channel (so mono) and it worked.  I could capture the Zoom speaker and send my whole daw output with a mix minus to Zoom w/out the callers output.  It's a bit of a hack since it's mono for each speaker but good enough to get going.     

If there were two BH devices one BH(2) and BH(16) I think this problem would go away since each software wouldn't be sharing the same output devices.   

Thanks for the great tool, can't wait to see more routing features :-)

**Comment by rmonge** [comment]
This article uses Loopback but is similar to what many folks want to do with BH https://razyalov.me/blog/mix-minus-logic-pro-x/  but without multiple BH devices it gets tricky.  Most applications don't let you set the channel for output/input and assume 1-2 so you end up needing multiple devices for output and input.

**Comment by rmonge** [comment]
This article uses Loopback but is similar to what many folks want to do with BH https://razyalov.me/blog/mix-minus-logic-pro-x/  but without multiple BH devices it gets tricky.  Most applications don't let you set the channel for output/input and assume 1-2 so you end up needing multiple devices for output and input.

**Comment by devinroth** [comment]
@lditti I hope this helps. https://existential.audio/howto/StreamFromLogicProXtoSkype.php

**Comment by devinroth** [comment]
@lditti I hope this helps. https://existential.audio/howto/StreamFromLogicProXtoSkype.php

**Comment by lditti** [comment]
Thanks @devinroth - I followed the instructions but the sound from logic was coming through my speakers and not my headphones with really bad feedback. I'm sure I was doing something wrong. Do you happen to know what it could have been?

![image](https://user-images.githubusercontent.com/64761155/83360705-3ed44a80-a351-11ea-8742-c1e38eda5bd1.png)

![image](https://user-images.githubusercontent.com/64761155/83360708-472c8580-a351-11ea-819a-1d96676b8882.png)

![image](https://user-images.githubusercontent.com/64761155/83360716-6b886200-a351-11ea-9ee4-e978b51d5afe.png)

**Comment by lditti** [comment]
Thanks @devinroth - I followed the instructions but the sound from logic was coming through my speakers and not my headphones with really bad feedback. I'm sure I was doing something wrong. Do you happen to know what it could have been?

![image](https://user-images.githubusercontent.com/64761155/83360705-3ed44a80-a351-11ea-8742-c1e38eda5bd1.png)

![image](https://user-images.githubusercontent.com/64761155/83360708-472c8580-a351-11ea-819a-1d96676b8882.png)

![image](https://user-images.githubusercontent.com/64761155/83360716-6b886200-a351-11ea-9ee4-e978b51d5afe.png)

**Comment by devinroth** [comment]
When your headphone are connected macOS is suppose to switch automatically to your headphones from your speakers. 

Please confirm that audio is playing through your headphones normally. In the picture it doesn’t look like they are connected.

**Comment by devinroth** [comment]
When your headphone are connected macOS is suppose to switch automatically to your headphones from your speakers. 

Please confirm that audio is playing through your headphones normally. In the picture it doesn’t look like they are connected.

**Comment by lditti** [comment]
@devinroth - sorry for the delay. See below - should I check off external headphones in the
multi-output device? And the audio is coming through my headphones.

<img width="632" alt="Screen Shot 2020-06-07 at 1 44 34 PM" src="https://user-images.githubusercontent.com/64761155/83976031-5a959e80-a8c5-11ea-9c8d-65598bc87560.png">

**Comment by lditti** [comment]
@devinroth - sorry for the delay. See below - should I check off external headphones in the
multi-output device? And the audio is coming through my headphones.

<img width="632" alt="Screen Shot 2020-06-07 at 1 44 34 PM" src="https://user-images.githubusercontent.com/64761155/83976031-5a959e80-a8c5-11ea-9c8d-65598bc87560.png">

**Comment by devinroth** [comment]
Yeah. Try replacing macbook pro speakers with your headphobes. Just remember to make sureBH is still 2nd on the list. 

Devin

**Comment by devinroth** [comment]
Yeah. Try replacing macbook pro speakers with your headphobes. Just remember to make sureBH is still 2nd on the list. 

Devin

**Comment by lditti** [comment]
Great - I recreated the entire process above and I'm not getting that terrible feedback any longer. I will let you know how it goes once I use it next Sunday. I'm using it to play the piano through Logic with my virtual skype piano lessons.

**Comment by lditti** [comment]
Great - I recreated the entire process above and I'm not getting that terrible feedback any longer. I will let you know how it goes once I use it next Sunday. I'm using it to play the piano through Logic with my virtual skype piano lessons.
